virtio-devices/src/balloon.rs

```diff
@@ -201,6 +203,30 @@ impl BalloonEpollHandler {
 
                 let gpa = (pfn as u64) << VIRTIO_BALLOON_PFN_SHIFT;
                 if let Ok(hva) = mem.get_host_address(GuestAddress(gpa)) {
+                    if queue_index == 0 {
```

Comment:
Don't use the queue index to check if you're in a case of `INFLATE`. Instead, you should do:
```rust
match ev_type {
    INFLATE_QUEUE_EVENT => {
        ...
    }
    _ => {}
}
```